### DISCUSSION TITLE
Dependency format and Maven central

### DISCUSSION THREAD

**Discussion by aghasemi** [jbangdev/jbang#1265] [Ideas]
Hi, 
In the docs, it is said that dependencies follow the "gradle-style locator" format. However, neither of the formats provided in Maven central can be used in JBang without change. Am I right? For example, for Gradle Short I have to manually remove `implementation` and the single equations.

This is another lazy-person's question maybe, but I just thought I may have missed something.

**Comment by quintesse** [comment]
Indeed. I think what the docs mean by Gradle-style is the `group:artifact:version` format with the colons separating the values.

I think you _can_ use the "Groovy Grape" ones exactly as mentioned though (but I've personally never used that format so I'm not sure).

**Reply by aghasemi** [reply]
Yes. But that requires an extra import statement, plus a `@GrabResolver` annotation, and I think one shall have only one `@Grapes` annotation (no?), so copy/paste from Maven central again breaks from second dependency onward.

The closest format to DEPS seems to be that of Buildr (e.g. `'junit:junit:jar:4.13.2'`), which has added only single quotes and a `:jar` identifier. Maybe JBang can accept this string as is, without one having to remove single quotes and  `:jar` substring manually?

**Reply by maxandersen** [reply]
If your issue is that search.maven.org does not show something directly copy/pastable then:

1) Wait for https://github.com/sonatype-nexus-community/search-maven-org/pull/137 to be merged (was approved back in 2021) then jbang format will show up
2) run `jbang app install gavsearch@jbangdev` and use `gavsearch hibernate-orm` to find GAV's and return i it in gradle,maven,jbang style formats.

about supporting :jar: format then I don't think bad UI on search.maven.org justifies the complexity it would bring.

**Reply by aghasemi** [reply]
> 1. Wait for [add jbang dependency option sonatype-nexus-community/search-maven-org#137](https://github.com/sonatype-nexus-community/search-maven-org/pull/137) to be merged (was approved back in 2021) then jbang format will show up

That's indeed the perfect solution!
